CLDR-13620 Add permillion alias to units.xml

sffc opened the pull request:
We had agreed to map "part-per-million" to "permillion", but this was not reflected in units.xml.

Reasons to replace this one:

1. "-per-" is a reserved syntax.
2. More consistent with "percent", "permille", and "permyriad".

##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/CLDR-13620
- [x] Updated PR title and link in previous line to include Issue number

sffc commented:
Are there other files that need to be updated?

I guess it's too late for CLDR 37.  We could open a bug and assign it to @hugovdm or @younies to do in CLDR 38?

macchiati commented:
Let's talk; part-per-million fits in BCP47 but permillion, perbillion, etc
don't.

Mark


On Sun, Feb 23, 2020 at 2:20 PM Shane F. Carr <notifications@github.com>
wrote:

> Are there other files that need to be updated?
>
> I guess it's too late for CLDR 37. We could open a bug and assign it to
> @hugovdm <https://github.com/hugovdm> or @younies
> <https://github.com/younies> to do in CLDR 38?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/unicode-org/cldr/pull/353?email_source=notifications&email_token=ACJLEMCN76VCTN32MRX33FDRELZBHA5CNFSM4KZ6KZ2KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEMWJBDI#issuecomment-590123149>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACJLEMEAHLYTZPMSG5FJ2HTRELZBHANCNFSM4KZ6KZ2A>
> .
>

sffc commented:
Acknowledged that permillion isn't in `\w{3,8}`.  The other reasons for changing listed in the PR description, which we've discussed on multiple occasions, still stand.  Given that the other proportion units also don't all fit in `\w{3,8}`, we aren't creating any new entropy.

Also: I added to this PR changes that make all the quantities be single components.

macchiati commented:
I am rethinking this further, so we should discuss. The quantity values are purely informative, and we should not require them to be in any particular format.

sffc commented:
OK, as discussed, I reverted the part changing the unit quantity names, and I fixed the unit test with the permillion alias.  I also did like in #359 where I updated permillion throughout CLDR.  PTAL.

macchiati commented:
What I wanted was to have separate PRs for:
1.  simple change of changing just the units.xml and unit test files, eg adding permillion to the unitAliases and changing other spots in that file
**That we could do right now**

2. the global "change all translations to use unaliased ids"
**That we need to wait on clearance until Monday**

sffc commented:
Ok, I moved the part about applying the permillion aliases to #363.
